src/dataflow/components/dataflow-program.tsx

```diff
@@ -322,7 +328,7 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
 
       this.updateRunAndGraphStates();
 
-      if (!this.props.readOnly && !this.isComplete()) {
+      if (!this.props.readOnly && !this.isComplete() || this.props.programIsRunning === "true") {
```

Comment:
Is there an extra case that `this.props.programIsRunning === "true"` handles?

Reply:
I think this helps identify if the program had not yet been run - without digging back in I can't recall exactly but it seems to work...